compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -929,13 +929,14 @@ impl<'a> Resolver<'a> {
         let name_ident = HirIdent::non_trait_method(id, location);
 
         let attributes = func.attributes().clone();
+        let should_fold = attributes.is_foldable();
 
         let mut generics = vecmap(&self.generics, |(_, typevar, _)| typevar.clone());
         let mut parameters = vec![];
         let mut parameter_types = vec![];
 
         for Param { visibility, pattern, typ, span: _ } in func.parameters().iter().cloned() {
-            if visibility == Visibility::Public && !self.pub_allowed(func) {
+            if visibility == Visibility::Public && !self.pub_allowed(func) && !should_fold {
```

Comment:
We do not want this warning as we will want to be able to specify public inputs on foldable functions

Reply:
`should_fold` should just be included as part of `pub_allowed` it seems. It looks like each place pub_allowed is used so is folding.